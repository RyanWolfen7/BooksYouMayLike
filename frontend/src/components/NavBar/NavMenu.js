import React from 'react'
import { withRouter } from 'react-router'
import { Menu, MenuItem } from '@material-ui/core'

const NavMenu = props => {
    const { menuAnchor, handleMenuClose, history } = props

    return (
        <Menu
            id={'NavMenu'}
            anchorEl={menuAnchor}
            anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
            keepMounted
            transformOrigin={{ vertical: 'top', horizontal: 'right' }}
            open={Boolean(menuAnchor)}
            onClose={handleMenuClose}
        >
            <MenuItem onClick={() => history.push('/addBook')}>
                Add Book To Your List
            </MenuItem>
        </Menu>
    )
}

export default withRouter(NavMenu)