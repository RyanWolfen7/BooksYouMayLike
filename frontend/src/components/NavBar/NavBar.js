import React, { useState } from 'react'
import { AppBar, Typography, InputBase, Toolbar, IconButton, Link } from '@material-ui/core'
import  SearchIcon from '@material-ui/icons/Search'
import MenuIcon from '@material-ui/icons/Menu';
import { NavStyles } from '../../styles/NavBarStyles'
import NavMenu from './NavMenu'

const NavBar = ({children}) => {
    const classes = NavStyles()
    const [menuAnchor, setMenuAnchor] = useState(null)

    const handleMenuClick = event => {
        setMenuAnchor(event.currentTarget)
    }

    const handleMenuClose = () => {
        setMenuAnchor(null)
    }

    return (
        <div className={classes.grow}>
            <AppBar position='static' className={classes.bar}>
                <Toolbar>
                    <Typography variant='h5' className={classes.title} noWrap>
                        <Link className={classes.homeLink} href='https://www.perlego.com/' color='inherit'>
                            Perlego
                        </Link>
                    </Typography>
                    {/* Refactor to new component when implimenting Functionality */}
                    <div className={classes.search}>
                        <div className={classes.searchIcon}>
                            <SearchIcon/>
                        </div>
                        <InputBase
                            placeholder="Search..."
                            classes={{
                                root: classes.inputRoot,
                                input: classes.inputInput
                            }}
                            inputProps={{ 'aria-label': 'search' }}
                        />
                    </div>
                    <div className={classes.sectionDrawer}>
                    <IconButton
                        id='NavMenuButton' 
                        aria-label='open drawer'
                        aria-controls='NavMenu'
                        aria-haspopup='true'
                        onClick={handleMenuClick}
                    >
                        <MenuIcon/>
                    </IconButton>
                    </div>
                </Toolbar>
            </AppBar>
            {NavMenu({menuAnchor, handleMenuClose})}
        </div>
    )
}

export default NavBar