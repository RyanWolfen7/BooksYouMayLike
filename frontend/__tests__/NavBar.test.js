import React from 'react'
import { shallow, mount } from 'enzyme'
import NavBar from '../src/components/NavBar/NavBar'
import NavBarMenu from '../src/components/NavBar/NavMenu'

let nav
let menu

describe('Snapshot testing', () => {

    beforeAll(() => {
        nav = shallow(<NavBar />)
        menu = shallow(<NavBarMenu />)
    })
    
    test('Nav to match snapshot', () => {
        expect(nav).toMatchSnapshot()
    })
    
    test('NavMenue to match snapshot', () => {
        expect(menu).toMatchSnapshot()
    })
})

describe('Component Functionality', () => {

    // console.log(shallow(<NavBar/>).debug())
    // console.log(shallow(<NavBarMenu/>).debug())

    const handleMenuClick = jest.fn()
    const handleMenuClose = jest.fn()
})