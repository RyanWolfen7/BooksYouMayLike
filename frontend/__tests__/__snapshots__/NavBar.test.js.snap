// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Nav to match snapshot 1`] = `
<div
  className="makeStyles-grow-2"
>
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-bar-1"
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-4"
        noWrap={true}
        variant="h5"
      >
        Perlego
      </WithStyles(ForwardRef(Typography))>
      <div
        className="makeStyles-search-5"
      >
        <div
          className="makeStyles-searchIcon-6"
        >
          <SearchIcon />
        </div>
        <WithStyles(ForwardRef(InputBase))
          classes={
            Object {
              "input": "makeStyles-inputInput-8",
              "root": "makeStyles-inputRoot-7",
            }
          }
          inputProps={
            Object {
              "aria-label": "search",
            }
          }
          placeholder="Search..."
        />
      </div>
      <div
        className="makeStyles-sectionDrawer-9"
      >
        <WithStyles(ForwardRef(IconButton))
          aria-controls="NavMenu"
          aria-haspopup="true"
          aria-label="open drawer"
          id="NavMenuButton"
          onClick={[Function]}
        >
          <MenuIcon />
        </WithStyles(ForwardRef(IconButton))>
      </div>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Menu))
    anchorEl={null}
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    id="NavMenu"
    keepMounted={true}
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
  >
    <WithStyles(ForwardRef(MenuItem))>
      <h1>
        Hello
      </h1>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))>
      <h2>
        World
      </h2>
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Menu))>
</div>
`;

exports[`Snapshot testing NavMenue to match snapshot 1`] = `
<WithStyles(ForwardRef(Menu))
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  id="NavMenu"
  keepMounted={true}
  open={false}
  transformOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
>
  <WithStyles(ForwardRef(MenuItem))>
    <h1>
      Hello
    </h1>
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))>
    <h2>
      World
    </h2>
  </WithStyles(ForwardRef(MenuItem))>
</WithStyles(ForwardRef(Menu))>
`;
